Make debug subgraph inlining thread local

zdevito opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #19217 Make it possible for self.forward to return a ScriptMethod
* #19142 graph_for based on last_optimized_executed_graph
* #19141 Remove GraphExecutor's python bindings
* #19138 Cleanup ScriptModule bindings
* #19137 get propagate_shape logic out of module.h
* **#19136 Make debug subgraph inlining thread local**
* #19086 Support Kwargs in C++ Function/Method calls

Summary: The amount of places this function needed to be threaded through
was bad, and would get works with script::Function being exposed to python.

This turns it into a thread local property, and updates the tests
that need it. This requires a lot less argument threading to achieve
the same result.

Test Plan: test_jit.py

Differential Revision: [D14885051](https://our.internmc.facebook.com/intern/diff/D14885051)

zdevito committed:
Make debug subgraph inlining thread local

Summary: The amount of places this function needed to be threaded through
was bad, and would get works with script::Function being exposed to python.

This turns it into a thread local property, and updates the tests
that need it. This requires a lot less argument threading to achieve
the same result.

Test Plan: test_jit.py

zdevito committed:
Update on "Make debug subgraph inlining thread local"

Make debug subgraph inlining thread local

Summary: The amount of places this function needed to be threaded through
was bad, and would get works with script::Function being exposed to python.

This turns it into a thread local property, and updates the tests
that need it. This requires a lot less argument threading to achieve
the same result.

Test Plan: test_jit.py

gh-metadata: pytorch pytorch 19136 gh/zdevito/14/head

zdevito committed:
Update on "Make debug subgraph inlining thread local"

Make debug subgraph inlining thread local

Summary: The amount of places this function needed to be threaded through
was bad, and would get works with script::Function being exposed to python.

This turns it into a thread local property, and updates the tests
that need it. This requires a lot less argument threading to achieve
the same result.

Test Plan: test_jit.py

gh-metadata: pytorch pytorch 19136 gh/zdevito/14/head

zdevito committed:
Update on "Make debug subgraph inlining thread local"

Make debug subgraph inlining thread local

Summary: The amount of places this function needed to be threaded through
was bad, and would get works with script::Function being exposed to python.

This turns it into a thread local property, and updates the tests
that need it. This requires a lot less argument threading to achieve
the same result.

Test Plan: test_jit.py

gh-metadata: pytorch pytorch 19136 gh/zdevito/14/head

zdevito committed:
Update on "Make debug subgraph inlining thread local"

Make debug subgraph inlining thread local

Summary: The amount of places this function needed to be threaded through
was bad, and would get works with script::Function being exposed to python.

This turns it into a thread local property, and updates the tests
that need it. This requires a lot less argument threading to achieve
the same result.

Test Plan: test_jit.py

gh-metadata: pytorch pytorch 19136 gh/zdevito/14/head

suo reviewed (approved):
This change looks good to me. `add_autograd_test()` is getting precariously nested, but that's out of scope for this PR

zdevito committed:
Update on "Make debug subgraph inlining thread local"

Make debug subgraph inlining thread local

Summary: The amount of places this function needed to be threaded through
was bad, and would get works with script::Function being exposed to python.

This turns it into a thread local property, and updates the tests
that need it. This requires a lot less argument threading to achieve
the same result.

Test Plan: test_jit.py

gh-metadata: pytorch pytorch 19136 gh/zdevito/14/head

zdevito committed:
Update on "Make debug subgraph inlining thread local"

Make debug subgraph inlining thread local

Summary: The amount of places this function needed to be threaded through
was bad, and would get works with script::Function being exposed to python.

This turns it into a thread local property, and updates the tests
that need it. This requires a lot less argument threading to achieve
the same result.

Test Plan: test_jit.py

gh-metadata: pytorch pytorch 19136 gh/zdevito/14/head

zdevito committed:
Update on "Make debug subgraph inlining thread local"

Make debug subgraph inlining thread local

Summary: The amount of places this function needed to be threaded through
was bad, and would get works with script::Function being exposed to python.

This turns it into a thread local property, and updates the tests
that need it. This requires a lot less argument threading to achieve
the same result.

Test Plan: test_jit.py

gh-metadata: pytorch pytorch 19136 gh/zdevito/14/head

zdevito committed:
Update on "Make debug subgraph inlining thread local"

Make debug subgraph inlining thread local

Summary: The amount of places this function needed to be threaded through
was bad, and would get works with script::Function being exposed to python.

This turns it into a thread local property, and updates the tests
that need it. This requires a lot less argument threading to achieve
the same result.

Test Plan: test_jit.py

gh-metadata: pytorch pytorch 19136 gh/zdevito/14/head

zdevito committed:
Update on "Make debug subgraph inlining thread local"

Make debug subgraph inlining thread local

Summary: The amount of places this function needed to be threaded through
was bad, and would get works with script::Function being exposed to python.

This turns it into a thread local property, and updates the tests
that need it. This requires a lot less argument threading to achieve
the same result.

Test Plan: test_jit.py

gh-metadata: pytorch pytorch 19136 gh/zdevito/14/head

zdevito committed:
Update on "Make debug subgraph inlining thread local"

Make debug subgraph inlining thread local

Summary: The amount of places this function needed to be threaded through
was bad, and would get works with script::Function being exposed to python.

This turns it into a thread local property, and updates the tests
that need it. This requires a lot less argument threading to achieve
the same result.

Test Plan: test_jit.py

gh-metadata: pytorch pytorch 19136 gh/zdevito/14/head

zdevito committed:
Update on "Make debug subgraph inlining thread local"

Make debug subgraph inlining thread local

Summary: The amount of places this function needed to be threaded through
was bad, and would get works with script::Function being exposed to python.

This turns it into a thread local property, and updates the tests
that need it. This requires a lot less argument threading to achieve
the same result.

Test Plan: test_jit.py

gh-metadata: pytorch pytorch 19136 gh/zdevito/14/head
